documentation/src/docs/asciidoc/user-guide/writing-tests.adoc

```diff
@@ -1688,6 +1688,9 @@ this API, you can access the provided arguments through a single argument passed
 test method. In addition, type conversion is supported as discussed in
 <<writing-tests-parameterized-tests-argument-conversion-implicit>>.
 
+Additionally, `{ArgumentsAccessor}` makes the current index of the argument sources
```

Comment:
I'm not sure that the average user would know what "current index of the argument sources" means. So it's probably better to add a bit more prose here to briefly explain it.

Reply:
I've rewritten this to hopefully make it clearer.